diff --git a/hardware/nvidia/platform/t23x/concord/kernel-dts/Makefile b/hardware/nvidia/platform/t23x/concord/kernel-dts/Makefile
index 450191e..b22d812 100644
--- a/hardware/nvidia/platform/t23x/concord/kernel-dts/Makefile
+++ b/hardware/nvidia/platform/t23x/concord/kernel-dts/Makefile
@@ -23,6 +23,7 @@ dtb-$(BUILD_ENABLE) += tegra234-p3701-0000-as-p3767-0001-p3737-0000.dtb
 dtb-$(BUILD_ENABLE) += tegra234-p3701-0000-as-pxxxx-p3737-0000.dtb
 dtb-$(BUILD_ENABLE) += tegra234-p3701-0000-p3737-0000-kexec.dtb
 dtb-$(BUILD_ENABLE) += tegra234-p3701-0004-p3737-0000.dtb
+dtb-$(BUILD_ENABLE) += tegra234-p3701-0000-p3737-0000-camera-4lane-eimx485.dtb
 dtbo-$(BUILD_ENABLE) += tegra234-p3737-overlay.dtbo
 dtbo-$(BUILD_ENABLE) += tegra234-p3737-audio-codec-rt5658.dtbo
 dtbo-$(BUILD_ENABLE) += tegra234-p3737-audio-codec-rt5640.dtbo
diff --git a/hardware/nvidia/platform/t23x/concord/kernel-dts/cvb/tegra234-p3737-0000-camera-4lane-eimx485.dtsi b/hardware/nvidia/platform/t23x/concord/kernel-dts/cvb/tegra234-p3737-0000-camera-4lane-eimx485.dtsi
new file mode 100644
index 0000000..a3612b1
--- /dev/null
+++ b/hardware/nvidia/platform/t23x/concord/kernel-dts/cvb/tegra234-p3737-0000-camera-4lane-eimx485.dtsi
@@ -0,0 +1,1178 @@
+/* Econ device tree for IMX485 support */
+#include <dt-bindings/media/camera.h>
+#include "dt-bindings/clock/tegra234-clock.h"
+#include <t234-common-cvb/tegra234-p3711-display.dtsi>
+#define IO_EXP_RST	TEGRA234_MAIN_GPIO(AC, 1)
+#define MIPICFG0	TEGRA234_MAIN_GPIO(H, 3)
+#define MIPICFG1	TEGRA234_MAIN_GPIO(H, 6)
+#define MIPICFG2       TEGRA234_MAIN_GPIO(AC, 0)
+
+/ {
+	/* setting Mipi cfg GPIO for lane Configuration */
+	gpio@2200000 {
+		mipi-cfg-gpios {
+			gpio-hog;
+			output-low;
+			gpios = <MIPICFG0 1 MIPICFG1 1 MIPICFG2 0 IO_EXP_RST 1>;
+			label = "mipi_sel_1", "mipi_sel_2", "mipi_sel_3", "cam_ioexp_rst";
+			status = "okay";
+		};
+	};
+
+
+	/* all cameras are disabled by default */
+
+	capture_vi_base: tegra-capture-vi {
+			num-channels = <4>;
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "okay";
+			eimx485_vi_port0: port@0 {
+					reg = <0>;
+					status = "okay";
+					eimx485_vi_in0: endpoint {
+						status = "okay";
+						port-index = <0>;
+						bus-width = <4>;
+						remote-endpoint = <&eimx485_csi_out0>;
+					};
+				};
+			eimx485_vi_port1: port@1 {
+					reg = <1>;
+					status = "okay";
+					eimx485_vi_in1: endpoint {
+						status = "okay";
+						port-index = <2>;
+						bus-width = <4>;
+						remote-endpoint = <&eimx485_csi_out1>;
+					};
+				};
+			eimx485_vi_port2: port@2 {
+					reg = <2>;
+					status = "okay";
+					eimx485_vi_in2: endpoint {
+						port-index = <4>;
+						status = "okay";
+						bus-width = <4>;
+						remote-endpoint = <&eimx485_csi_out2>;
+					};
+				};
+			eimx485_vi_port3: port@3 {
+					reg = <3>;
+					status = "okay";
+					eimx485_vi_in3: endpoint {
+						port-index = <5>;
+						status = "okay";
+						bus-width = <4>;
+						remote-endpoint = <&eimx485_csi_out3>;
+					};
+				};
+			};
+		};
+
+
+	host1x@13e00000 {	
+		csi_base: nvcsi@15a00000 {
+			num-channels = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "okay";
+			csi_chan0: channel@0 {
+				reg = <0>;
+				status = "okay";
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						status = "okay";
+						eimx485_csi_in0: endpoint@0 {
+							status = "okay";
+							port-index = <0>;
+							bus-width = <4>;
+							remote-endpoint = <&eimx485_eimx485_out0>;
+						};
+					};
+					port@1 {
+						reg = <1>;
+						status = "okay";
+						eimx485_csi_out0: endpoint@1 {
+							status = "okay";
+							remote-endpoint = <&eimx485_vi_in0>;
+						};
+					};
+				};
+			};
+			csi_chan1: channel@1 {
+				reg = <1>;
+				status = "okay";
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						status = "okay";
+						eimx485_csi_in1: endpoint@2 {
+							status = "okay";
+							port-index = <2>;
+							bus-width = <4>;
+							remote-endpoint = <&eimx485_eimx485_out1>;
+						};
+					};
+					port@1 {
+						reg = <1>;
+						status = "okay";
+						eimx485_csi_out1: endpoint@3 {
+							status = "okay";
+							remote-endpoint = <&eimx485_vi_in1>;
+						};
+					};
+				};
+			};
+			csi_chan2: channel@2 {
+				reg = <2>;
+				status = "okay";
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						status = "okay";
+						eimx485_csi_in2: endpoint@4 {
+							status = "okay";
+							port-index = <4>;
+							bus-width = <4>;
+							remote-endpoint = <&eimx485_eimx485_out2>;
+						};
+					};
+					port@1 {
+						reg = <1>;
+						status = "okay";
+						eimx485_csi_out2: endpoint@5 {
+							status = "okay";
+							remote-endpoint = <&eimx485_vi_in2>;
+						};
+					};
+				};
+			};
+			csi_chan3: channel@3 {
+				reg = <3>;
+				status = "okay";
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						status = "okay";
+						eimx485_csi_in3: endpoint@6 {
+							status = "okay";
+							port-index = <6>;
+							bus-width = <4>;
+							remote-endpoint = <&eimx485_eimx485_out3>;
+						};
+					};
+					port@1 {
+						reg = <1>;
+						status = "okay";
+						eimx485_csi_out3: endpoint@7 {
+							status = "okay";
+							remote-endpoint = <&eimx485_vi_in3>;
+						};
+					};
+				};
+			};
+		};
+	};
+	i2c@c240000{
+	//i2c@c250000
+	//i2c@31e0000 { 
+		status = "okay";
+		#address-cells = <1>;
+		#size-cells = <0>;	
+		tca6424_22: tca6424@22 {
+			compatible = "ti,tca6424";
+			gpio-controller;
+			#gpio-cells = <2>;
+			reg = <0x22>;
+                        vcc-supply = <&p3737_vdd_1v8_sys>;
+			status = "okay";
+			tca6424_22_outlow {
+				/*
+				 * GPIO-0  : RST_CAM1
+				 * GPIO-1  : RST_CAM2
+				 * GPIO-2  : RST_CAM3
+				 * GPIO-3  : RST_CAM4
+				 * GPIO-4  : RST_CAM5
+				 * GPIO-5  : RST_CAM6
+				 * GPIO-6  : PWDN_CAM1
+				 * GPIO-7  : PWDN_CAM2
+				 * GPIO-8  : PWDN_CAM3
+				 * GPIO-9  : PWDN_CAM4
+				 * GPIO-10 : PWDN_CAM5
+				 * GPIO-11 : PWDN_CAM6
+				 */
+
+				gpio-hog;
+				gpios = <0 0 1 0 2 0 3 0 4 0 5 0 6 0 7 0 8 0 9 0 10 0 11 0 12 0 13 0 14 0 15 0 22 0 23 0>; 
+				output-low;
+				label = "tca6424_22_reset-cam0_0",
+					"tca6424_22_reset-cam1_1",
+					"tca6424_22_reset-cam2_2",
+					"tca6424_22_reset-cam3_3",
+					"tca6424_22_reset-cam4_4",
+					"tca6424_22_reset-cam5_5",
+					"tca6424_22_boot-cam0_6",
+					"tca6424_22_boot-cam1_7",
+					"tca6424_22_boot-cam2_8",
+					"tca6424_22_boot-cam3_9",
+					"tca6424_22_boot-cam4_10",
+					"tca6424_22_boot-cam5_11",
+					"tca6424_22_flash-cam0_12",
+					"tca6424_22_flash-cam1_13";
+			};
+			tca6424_22_outhigh {
+				status= "disabled";
+			};
+			tca6424_22_input {
+			//	status = "disabled";
+				gpio-hog;
+				gpios = < 16 0 17 0 18 0 19 0 20 0 21 0>;
+				output-low;
+				label = "tca6424_22_spi_cam0_16",
+					  "tca6424_22_spi_cam1_17",
+					  "tca6424_22_spi_cam2_18",
+					  "tca6424_22_spi_cam3_19",
+					  "tca6424_22_spi_cam4_20",
+					  "tca6424_22_spi_cam5_21";
+			};
+		};
+	};
+
+	i2c@3180000 {
+		status = "okay";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		tca9548@70 {
+                        compatible = "nxp,pca9548";
+                        reg = <0x70>;
+                        #address-cells = <1>;
+                        #size-cells = <0>;
+                        vcc-supply = <&p3737_vdd_1v8_sys>;
+                        skip_mux_detect;
+                        force_bus_start = <0x1e>;
+			i2c@0 {
+                                reg = <0>;
+                                i2c-mux,deselect-on-exit;
+                                #address-cells = <1>;
+                                #size-cells = <0>;
+                                status = "okay";
+				eimx485_a@42 {
+				    status = "okay";
+				    compatible = "nvidia,eimx485";
+				    reg = <0x42>;
+				    devnode = "video0";
+				    sensor_model ="eimx485";
+				    physical_w = "5.60";
+				    physical_h = "3.175";
+				    set_mode_delay_ms = "1000";
+				    //post_crop_frame_drop = "0";
+				    use_decibel_gain = "true";
+				    use_sensor_mode_id = "true";
+				    use_dol_wdr_mode = "false";
+				    use_master_slave_mode = "true";
+				    limit_analog_gain = "true";
+				    camera_mipi_lanes = "4";
+				    avdd-reg = "vana";
+				    iovdd-reg = "vif";
+				    clocks = <&bpmp_clks TEGRA234_CLK_EXTPERIPH1>, <&bpmp_clks TEGRA234_CLK_PLLP_OUT0>;
+				    clock-names = "extperiph1", "pllp_grtba";
+				    mclk = "extperiph1";
+				    reset-gpios = <&tca6424_22 0 GPIO_ACTIVE_HIGH>;
+				    flash-gpios = <&tca6424_22 12 GPIO_ACTIVE_HIGH>;
+				    boot-gpios =  <&tca6424_22 6 GPIO_ACTIVE_HIGH>;
+				    mipicfg1-gpios = <&tegra_main_gpio MIPICFG0 GPIO_ACTIVE_HIGH>;
+                    		    mipicfg2-gpios = <&tegra_main_gpio MIPICFG1 GPIO_ACTIVE_HIGH>;
+				    master-select = /bits/ 16 <240>;
+				    vana-supply = <&p3737_avdd_cam_2v8>;
+				    vif-supply = <&p3737_vdd_1v8_sys>;
+				mode0 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_a";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "12";
+					csi_pixel_bit_depth = "12";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "3840";
+					active_h = "2160";
+					readout_orientation = "0";
+					line_length = "650";
+					inherent_gain = "1";
+					mclk_multiplier = "5.13";
+					pix_clk_hz = "190771200";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "50000000";
+					step_framerate = "1";
+					default_framerate = "50000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";
+					embedded_metadata_height = "1";
+				    };
+				mode1 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_a";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "10";
+					csi_pixel_bit_depth = "10";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "3840";
+					active_h = "2160";
+					readout_orientation = "0";
+					line_length = "550";
+					inherent_gain = "1";
+					mclk_multiplier = "7.39";
+					pix_clk_hz = "274710528";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "60000000";
+					step_framerate = "1";
+					default_framerate = "60000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";
+					embedded_metadata_height = "1";
+				    };
+				mode2 {
+					mclk_khz = "37125"; 
+					num_lanes = "4"; 
+					tegra_sinterface = "serial_a"; 
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "12";
+					csi_pixel_bit_depth = "12";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "1920";
+					active_h = "1080";
+					readout_orientation = "0";
+					line_length = "366";
+					inherent_gain = "1";
+					mclk_multiplier = "12.92";
+					pix_clk_hz = "480000000";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";	
+					max_hdr_ratio = "1";	
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "90000000";
+					step_framerate = "1";
+					default_framerate = "90000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";/* us */
+					embedded_metadata_height = "1";
+				};
+				mode3 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_a";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "10";
+					csi_pixel_bit_depth = "10";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "1920";
+					active_h = "1080";
+					readout_orientation = "0";
+					line_length = "366";
+					inherent_gain = "1";
+					mclk_multiplier = "15.51";
+					pix_clk_hz = "576000000"; 
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "90000000";
+					step_framerate = "1";
+					default_framerate = "90000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";/* us */
+					embedded_metadata_height = "1";
+				};
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						eimx485_eimx485_out0: endpoint {
+							port-index = <0>;
+							bus-width = <4>;
+							remote-endpoint = <&eimx485_csi_in0>;
+						};
+					};
+				  };
+			     };
+			};
+			i2c@2 {
+                                reg = <2>;
+                                i2c-mux,deselect-on-exit;
+                                #address-cells = <1>;
+                                #size-cells = <0>;
+                                status = "okay";
+				eimx485_c@42 {
+				    status = "okay";
+				    compatible = "nvidia,eimx485";
+				    reg = <0x42>;
+				    devnode = "video1";
+				    sensor_model ="eimx485";
+				    physical_w = "5.60";
+				    physical_h = "3.175";
+				    set_mode_delay_ms = "1000";
+				    //post_crop_frame_drop = "0";
+				    use_decibel_gain = "true";
+				    use_sensor_mode_id = "true";
+				    use_dol_wdr_mode = "false";
+				    use_master_slave_mode = "true";
+				    limit_analog_gain = "true";
+				    camera_mipi_lanes = "4";
+				    avdd-reg = "vana";
+				    iovdd-reg = "vif";
+				   clocks = <&bpmp_clks TEGRA234_CLK_EXTPERIPH1>,
+							 <&bpmp_clks TEGRA234_CLK_PLLP_OUT0>;
+					clock-names = "extperiph1", "pllp_grtba";
+					mclk = "extperiph1";
+				    reset-gpios = <&tca6424_22 2 GPIO_ACTIVE_HIGH>;
+				    flash-gpios = <&tca6424_22 14 GPIO_ACTIVE_HIGH>;
+				    boot-gpios =  <&tca6424_22 8 GPIO_ACTIVE_HIGH>;
+				    mipicfg1-gpios = <&tegra_main_gpio MIPICFG0 GPIO_ACTIVE_HIGH>;
+                    		    mipicfg2-gpios = <&tegra_main_gpio MIPICFG1 GPIO_ACTIVE_HIGH>;
+				    master-select = /bits/ 16 <242>;
+				    vana-supply = <&p3737_avdd_cam_2v8>;
+				    vif-supply = <&p3737_vdd_1v8_sys>;
+				mode0 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_c";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "12";
+					csi_pixel_bit_depth = "12";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "3840";
+					active_h = "2160";
+					readout_orientation = "0";
+					line_length = "650";
+					inherent_gain = "1";
+					mclk_multiplier = "5.13";
+					pix_clk_hz = "190771200";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "50000000";
+					step_framerate = "1";
+					default_framerate = "50000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";
+					embedded_metadata_height = "1";
+				    };
+				mode1 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_c";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "10";
+					csi_pixel_bit_depth = "10";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "3840";
+					active_h = "2160";
+					readout_orientation = "0";
+					line_length = "550";
+					inherent_gain = "1";
+					mclk_multiplier = "7.39";
+					pix_clk_hz = "274710528";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "60000000";
+					step_framerate = "1";
+					default_framerate = "60000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";
+					embedded_metadata_height = "1";
+				    };
+
+				mode2 {
+					mclk_khz = "37125"; 
+					num_lanes = "4"; 
+					tegra_sinterface = "serial_c"; 
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "12";
+					csi_pixel_bit_depth = "12";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "1920";
+					active_h = "1080";
+					readout_orientation = "0";
+					line_length = "366";
+					inherent_gain = "1";
+					mclk_multiplier = "12.92";
+					pix_clk_hz = "480000000";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";	
+					max_hdr_ratio = "1";	
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "90000000";
+					step_framerate = "1";
+					default_framerate = "90000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";/* us */
+					embedded_metadata_height = "1";
+				};
+				mode3 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_c";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "10";
+					csi_pixel_bit_depth = "10";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "1920";
+					active_h = "1080";
+					readout_orientation = "0";
+					line_length = "366";
+					inherent_gain = "1";
+					mclk_multiplier = "15.51";
+					pix_clk_hz = "576000000"; 
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "90000000";
+					step_framerate = "1";
+					default_framerate = "90000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";/* us */
+					embedded_metadata_height = "1";
+				};
+	
+
+	
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						eimx485_eimx485_out1: endpoint {
+							port-index = <2>;
+							bus-width = <4>;
+							remote-endpoint = <&eimx485_csi_in1>;
+						};
+					};
+				    };
+				};
+			};
+			i2c@4 {
+                                reg = <4>;
+                                i2c-mux,deselect-on-exit;
+                                #address-cells = <1>;
+                                #size-cells = <0>;
+                                status = "okay";
+				eimx485_e@42 {
+				    status = "okay";
+				    compatible = "nvidia,eimx485";
+				    reg = <0x42>;
+				    devnode = "video2";
+				    sensor_model ="eimx485";
+				    physical_w = "5.60";
+				    physical_h = "3.175";
+				    set_mode_delay_ms = "1000";
+				    //post_crop_frame_drop = "0";
+				    use_decibel_gain = "true";
+				    use_sensor_mode_id = "true";
+				    use_dol_wdr_mode = "false";
+				    use_master_slave_mode = "true";
+				    limit_analog_gain = "true";
+				    camera_mipi_lanes = "4";
+				    avdd-reg = "vana";
+				    iovdd-reg = "vif";
+				    clocks = <&bpmp_clks TEGRA234_CLK_EXTPERIPH1>,
+							 <&bpmp_clks TEGRA234_CLK_PLLP_OUT0>;
+				    clock-names = "extperiph1", "pllp_grtba";
+				    mclk = "extperiph1";
+				    reset-gpios = <&tca6424_22 4 GPIO_ACTIVE_HIGH>;
+				    flash-gpios = <&tca6424_22 23 GPIO_ACTIVE_HIGH>;
+				    boot-gpios =  <&tca6424_22 10 GPIO_ACTIVE_HIGH>;
+				    mipicfg1-gpios = <&tegra_main_gpio MIPICFG0 GPIO_ACTIVE_HIGH>;
+				    mipicfg2-gpios = <&tegra_main_gpio MIPICFG1 GPIO_ACTIVE_HIGH>;
+				    master-select = /bits/ 16 <244>;
+				    vana-supply = <&p3737_avdd_cam_2v8>;
+				    vif-supply = <&p3737_vdd_1v8_sys>;
+				mode0 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_e";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "12";
+					csi_pixel_bit_depth = "12";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "3840";
+					active_h = "2160";
+					readout_orientation = "0";
+					line_length = "650";
+					inherent_gain = "1";
+					mclk_multiplier = "5.13";
+					pix_clk_hz = "190771200";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "50000000";
+					step_framerate = "1";
+					default_framerate = "50000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";
+					embedded_metadata_height = "1";
+				    };
+				mode1 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_e";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "10";
+					csi_pixel_bit_depth = "10";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "3840";
+					active_h = "2160";
+					readout_orientation = "0";
+					line_length = "550";
+					inherent_gain = "1";
+					mclk_multiplier = "7.39";
+					pix_clk_hz = "274710528";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "60000000";
+					step_framerate = "1";
+					default_framerate = "60000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";
+					embedded_metadata_height = "1";
+				    };
+
+				mode2 {
+					mclk_khz = "37125"; 
+					num_lanes = "4"; 
+					tegra_sinterface = "serial_e"; 
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "12";
+					csi_pixel_bit_depth = "12";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "1920";
+					active_h = "1080";
+					readout_orientation = "0";
+					line_length = "366";
+					inherent_gain = "1";
+					mclk_multiplier = "12.92";
+					pix_clk_hz = "480000000";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";	
+					max_hdr_ratio = "1";	
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "90000000";
+					step_framerate = "1";
+					default_framerate = "90000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";/* us */
+					embedded_metadata_height = "1";
+				};
+				mode3 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_e";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "10";
+					csi_pixel_bit_depth = "10";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "1920";
+					active_h = "1080";
+					readout_orientation = "0";
+					line_length = "366";
+					inherent_gain = "1";
+					mclk_multiplier = "15.51";
+					pix_clk_hz = "576000000"; 
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "90000000";
+					step_framerate = "1";
+					default_framerate = "90000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";/* us */
+					embedded_metadata_height = "1";
+				};
+	
+
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						eimx485_eimx485_out2: endpoint {
+							port-index = <4>;
+							bus-width = <4>;
+							remote-endpoint = <&eimx485_csi_in2>;
+						};
+					};
+				    };
+				};
+			};
+			i2c@5 {
+                                reg = <5>;
+                                i2c-mux,deselect-on-exit;
+                                #address-cells = <1>;
+                                #size-cells = <0>;
+                                status = "okay";
+				eimx485_g@42 {
+				    status = "okay";
+				    compatible = "nvidia,eimx485";
+				    reg = <0x42>;
+				    devnode = "video3";
+				    sensor_model ="eimx485";
+				    physical_w = "5.60";
+				    physical_h = "3.175";
+				    set_mode_delay_ms = "1000";
+				    //post_crop_frame_drop = "0";
+				    use_decibel_gain = "true";
+				    use_sensor_mode_id = "true";
+				    use_dol_wdr_mode = "false";
+				    use_master_slave_mode = "true";
+				    limit_analog_gain = "true";
+				    camera_mipi_lanes = "4";
+				    avdd-reg = "vana";
+				    iovdd-reg = "vif";
+				    clocks = <&bpmp_clks TEGRA234_CLK_EXTPERIPH1>,
+							 <&bpmp_clks TEGRA234_CLK_PLLP_OUT0>;
+					clock-names = "extperiph1", "pllp_grtba";
+					mclk = "extperiph1";
+				    reset-gpios = <&tca6424_22 5 GPIO_ACTIVE_HIGH>;
+				    flash-gpios = <&tca6424_22 22 GPIO_ACTIVE_HIGH>;
+				    boot-gpios =  <&tca6424_22 11 GPIO_ACTIVE_HIGH>;
+				    mipicfg1-gpios = <&tegra_main_gpio MIPICFG0 GPIO_ACTIVE_HIGH>;
+                    		    mipicfg2-gpios = <&tegra_main_gpio MIPICFG1 GPIO_ACTIVE_HIGH>;
+				    master-select = /bits/ 16 <245>;
+				    vana-supply = <&p3737_avdd_cam_2v8>;
+				    vif-supply = <&p3737_vdd_1v8_sys>;
+				mode0 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_g";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "12";
+					csi_pixel_bit_depth = "12";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "3840";
+					active_h = "2160";
+					readout_orientation = "0";
+					line_length = "650";
+					inherent_gain = "1";
+					mclk_multiplier = "5.13";
+					pix_clk_hz = "190771200";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "50000000";
+					step_framerate = "1";
+					default_framerate = "50000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";
+					embedded_metadata_height = "1";
+				    };
+				mode1 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_g";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "10";
+					csi_pixel_bit_depth = "10";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "3840";
+					active_h = "2160";
+					readout_orientation = "0";
+					line_length = "550";
+					inherent_gain = "1";
+					mclk_multiplier = "7.39";
+					pix_clk_hz = "274710528";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "60000000";
+					step_framerate = "1";
+					default_framerate = "60000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";
+					embedded_metadata_height = "1";
+				    };
+				mode2 {
+					mclk_khz = "37125"; 
+					num_lanes = "4"; 
+					tegra_sinterface = "serial_g"; 
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "12";
+					csi_pixel_bit_depth = "12";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "1920";
+					active_h = "1080";
+					readout_orientation = "0";
+					line_length = "366";
+					inherent_gain = "1";
+					mclk_multiplier = "12.92";
+					pix_clk_hz = "480000000";
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";	
+					max_hdr_ratio = "1";	
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "90000000";
+					step_framerate = "1";
+					default_framerate = "90000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";/* us */
+					embedded_metadata_height = "1";
+				};
+				mode3 {
+					mclk_khz = "37125";
+					num_lanes = "4";
+					tegra_sinterface = "serial_g";
+					phy_mode = "DPHY";
+					discontinuous_clk = "no";
+					dpcm_enable = "false";
+					cil_settletime = "0";
+					dynamic_pixel_bit_depth = "10";
+					csi_pixel_bit_depth = "10";
+					mode_type = "bayer";
+					pixel_phase = "rggb";
+
+					active_w = "1920";
+					active_h = "1080";
+					readout_orientation = "0";
+					line_length = "366";
+					inherent_gain = "1";
+					mclk_multiplier = "15.51";
+					pix_clk_hz = "576000000"; 
+
+					gain_factor = "10";
+					min_gain_val = "0"; 
+					max_gain_val = "300"; 
+					step_gain_val = "3"; 
+					default_gain = "0";
+					min_hdr_ratio = "1";
+					max_hdr_ratio = "1";
+					framerate_factor = "1000000";
+					min_framerate = "2500000";
+					max_framerate = "90000000";
+					step_framerate = "1";
+					default_framerate = "90000000";
+					exposure_factor = "1000000";
+					min_exp_time = "450";
+					max_exp_time = "400000";
+					step_exp_time = "1";
+					default_exp_time = "16667";/* us */
+					embedded_metadata_height = "1";
+				};
+	
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						eimx485_eimx485_out3: endpoint {
+							port-index = <6>;
+							bus-width = <4>;
+							remote-endpoint = <&eimx485_csi_in3>;
+						};
+					};
+				};
+			    };
+			};
+		};
+	};
+};
+/{
+	 tegra-camera-platform {
+		compatible = "nvidia, tegra-camera-platform";
+		num_csi_lanes = <16>;
+		max_lane_speed = <1500000>;
+		min_bits_per_pixel = <10>;
+		vi_peak_byte_per_pixel = <2>;
+		vi_bw_margin_pct = <25>;
+		max_pixel_rate = <750000>;
+		isp_peak_byte_per_pixel = <5>;
+		isp_bw_margin_pct = <25>;
+		modules {
+			module0 {
+				badge = "eimx485_bottomleft_0485";
+				position = "bottomleft";
+				status = "okay";
+				orientation = "1";
+				drivernode0 {
+					pcl_id = "v4l2_sensor";
+					devname = "e-con_cam 30-0042";
+					proc-device-tree = "/proc/device-tree/i2c@3180000/tca9548@70/i2c@0/eimx485_a@42";
+					status = "okay";
+				};
+			};
+			module1 {
+				badge = "eimx485_centerleft_0485";
+				position = "centerleft";
+				status = "okay";
+				orientation = "1";
+				drivernode0 {
+					pcl_id = "v4l2_sensor";
+					devname = "e-con_cam 32-0042";
+					proc-device-tree = "/proc/device-tree/i2c@3180000/tca9548@70/i2c@2/eimx485_c@42";
+					status = "okay";
+				};
+			};
+			module2 {
+				badge = "eimx485_centerright_0485";
+				position = "centerright";
+				status = "okay";
+				orientation = "1";
+				drivernode0 {
+					pcl_id = "v4l2_sensor";
+					devname = "e-con_cam 34-0042";
+					proc-device-tree = "/proc/device-tree/i2c@3180000/tca9548@70/i2c@4/eimx485_e@42";
+					status = "okay";
+				};
+			};
+			module3 {
+				badge = "eimx485_topleft_0485";
+				position = "topleft";
+				status = "okay";
+				orientation = "1";
+				drivernode0 {
+					pcl_id = "v4l2_sensor";
+					devname = "e-con_cam 35-0042";
+					proc-device-tree = "/proc/device-tree/i2c@3180000/tca9548@70/i2c@5/eimx485_g@42";
+					status = "okay";
+				};
+			};
+		};
+	};
+};
+
diff --git a/hardware/nvidia/platform/t23x/concord/kernel-dts/tegra234-p3701-0000-p3737-0000-camera-4lane-eimx485.dts b/hardware/nvidia/platform/t23x/concord/kernel-dts/tegra234-p3701-0000-p3737-0000-camera-4lane-eimx485.dts
new file mode 100644
index 0000000..132e83d
--- /dev/null
+++ b/hardware/nvidia/platform/t23x/concord/kernel-dts/tegra234-p3701-0000-p3737-0000-camera-4lane-eimx485.dts
@@ -0,0 +1,126 @@
+/*
+ * tegra194-p2888-0001-p2822-0000-camera-ar1335.dts : DTS file for galen
+ * to enable ar1335 image sensor.
+ *
+ * Copyright (c) 2016, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ */
+#include "tegra234-p3701-0000-p3737-0000.dts"
+#include "cvb/tegra234-p3737-0000-camera-4lane-eimx485.dtsi"
+
+/ {
+	nvidia,dtsfilename = __FILE__;
+
+		capture_vi_base: tegra-capture-vi {
+				status = "okay";
+			ports {
+				eimx485_vi_port0: port@0 {
+					status = "okay";
+					eimx485_vi_in0: endpoint {
+						status = "okay";
+					};
+				};
+
+				eimx485_vi_port1: port@1 {
+					status = "okay";
+					eimx485_vi_in1: endpoint {
+						status = "okay";
+					};
+				};
+				eimx485_vi_port2: port@2 {
+					status = "okay";
+					eimx485_vi_in2: endpoint {
+						status = "okay";
+					};
+				};
+				eimx485_vi_port3: port@3 {
+					status = "okay";
+					eimx485_vi_in3: endpoint {
+						status = "okay";
+					};
+				};
+			};
+		};
+	host1x@13e00000 {
+
+		csi_base: nvcsi@15a00000 {
+			csi_chan0: channel@0 {
+				status = "okay";
+				ports {
+					csi_chan0_port0: port@0 {
+						status = "okay";
+						eimx485_csi_in0: endpoint@0 {
+							status = "okay";
+						};
+					};
+					csi_chan0_port1: port@1 {
+						status = "okay";
+						eimx485_csi_out0: endpoint@1 {
+							status = "okay";
+						};
+					};
+				};
+			};
+			csi_chan1: channel@1 {
+				status = "okay";
+				ports {
+					csi_chan1_port0: port@0 {
+						status = "okay";
+						eimx485_csi_in1: endpoint@2 {
+							status = "okay";
+						};
+					};
+					csi_chan1_port1: port@1 {
+						status = "okay";
+						eimx485_csi_out1: endpoint@3 {
+							status = "okay";
+						};
+					};
+				};
+			};
+			csi_chan2: channel@2 {
+				status = "okay";
+				ports {
+					csi_chan2_port0: port@0 {
+						status = "okay";
+						eimx485_csi_in2: endpoint@4 {
+							status = "okay";
+						};
+					};
+					csi_chan2_port1: port@1 {
+						status = "okay";
+						eimx485_csi_out2: endpoint@5 {
+							status = "okay";
+						};
+					};
+				};
+			};
+			csi_chan3: channel@3 {
+				status = "okay";
+				ports {
+					csi_chan3_port0: port@0 {
+						status = "okay";
+						eimx485_csi_in3: endpoint@6 {
+							status = "okay";
+						};
+					};
+					csi_chan3_port1: port@1 {
+						status = "okay";
+						eimx485_csi_out3: endpoint@7 {
+							status = "okay";
+						};
+					};
+				};
+			};
+		};
+	};
+};
